lnwallet/chanfunding/coin_select.go

```diff
@@ -123,28 +123,55 @@ func CoinSelect(feeRate chainfee.SatPerKWeight, amt btcutil.Amount,
 			return nil, 0, err
 		}
 
-		// Obtain fee estimate assuming a change output.
-		_, requiredFee, err := calculateFees(selectedUtxos, feeRate)
+		// Obtain fee estimates both with and without using a change
+		// output.
+		requiredFeeNoChange, requiredFeeWithChange, err := calculateFees(
+			selectedUtxos, feeRate)
 		if err != nil {
 			return nil, 0, err
 		}
+
 		// The difference between the selected amount and the amount
 		// requested will be used to pay fees, and generate a change
 		// output with the remaining.
 		overShootAmt := totalSat - amt
 
+		var changeAmt btcutil.Amount
+
+		switch {
+
 		// Based on the estimated size and fee rate, if the excess
 		// amount isn't enough to pay fees, then increase the requested
 		// coin amount by the estimate required fee, performing another
 		// round of coin selection.
-		if overShootAmt < requiredFee {
-			amtNeeded = amt + requiredFee
+		case overShootAmt < requiredFeeNoChange:
```

Comment:
update comment for the case to refleclt this is now without change output.

Reply:
Comment changed to make it more clear!